rasa/core/featurizers/single_state_featurizer.py

```diff
@@ -40,127 +102,227 @@ class SingleStateFeaturizer:
+            interpreter = RegexInterpreter()
 
-        self._default_feature_states[INTENT] = convert_to_dict(domain.intents)
-        self._default_feature_states[ACTION_NAME] = convert_to_dict(
-            domain.action_names_or_texts
+        # featurize the message/sub_state...
+        # TODO: is it clear that this works? if not, it should be made explicit
+        # somewhere somehow .... -> SubState and Message dataclass things....
+        message = Message(data=sub_state)
+        parsed_message = interpreter.featurize_message(message)  # TODO:
+
+        # ... and gather all features except the ones for Entities
+        exclude = exclude or []
+        attributes = set(
+            attribute for attribute in sub_state.keys() if attribute not in exclude
         )
-        self._default_feature_states[ENTITIES] = convert_to_dict(domain.entity_states)
-        self._default_feature_states[SLOTS] = convert_to_dict(domain.slot_states)
-        self._default_feature_states[ACTIVE_LOOP] = convert_to_dict(domain.form_names)
-        self.action_texts = domain.action_texts
-        self.entity_tag_specs = self._create_entity_tag_specs(bilou_tagging)
-
-    def _state_features_for_attribute(
-        self, sub_state: SubState, attribute: Text
-    ) -> Dict[Text, int]:
-        # FIXME: the code below is not type-safe, but fixing it
-        #        would require more refactoring, for instance using
-        #        data classes in our states
-        if attribute in {INTENT, ACTION_NAME}:
-            return {sub_state[attribute]: 1}  # type: ignore[dict-item]
-        elif attribute == ENTITIES:
-            return {entity: 1 for entity in sub_state.get(ENTITIES, [])}
-        elif attribute == ACTIVE_LOOP:
-            return {sub_state["name"]: 1}  # type: ignore[dict-item]
-        elif attribute == SLOTS:
-            return {
-                f"{slot_name}_{i}": value
-                for slot_name, slot_as_feature in sub_state.items()
-                for i, value in enumerate(slot_as_feature)
-            }
-        else:
-            raise ValueError(
-                f"Given attribute '{attribute}' is not supported. "
-                f"It must be one of '{self._default_feature_states.keys()}'."
-            )
+        attribute_to_features: Dict[Text, List[Features]] = dict()
+        for attribute in attributes:
+            all_features = parsed_message.get_sparse_features(
+                attribute
+            ) + parsed_message.get_dense_features(attribute)
 
-    def _create_features(
+            for features in all_features:
+                if features is not None:
+                    attribute_to_features[attribute].append(features)
+
+        attribute_list = keep_only_sparse_seq_converted_to_sent or []
+        for attribute in attribute_list:
+            if attribute_to_features.get(attribute):
+                converted_features = convert_sparse_sequence_to_sentence_features(
```

Comment:
As mentioned above, it would be good to have a bit more flexibility here (not just `sum`).

Reply:
I think that is a good idea and I think there are multiple places we could reuse that (cf. my funny countvectorizer related questions :)